xbout/plotting/animate.py

```diff
@@ -9,25 +9,84 @@
 from matplotlib.animation import PillowWriter
 
 
+if (
+    "pcolor.shading" in matplotlib.rcParams
+    and matplotlib.rcParams["pcolor.shading"] == "flat"
+):
+    # "flat" was the old matplotlib default which discarded the last row and column if
+    # X, Y and Z were all equal in size. The new "auto" should be better. Need to set
+    # this explicitly because "flat" is the default during a deprecation cycle.
+    matplotlib.rcParams["pcolor.shading"] = "auto"
+
+
+def _parse_coord_option(coord, axis_coords, da):
+    if isinstance(axis_coords, dict):
+        option_value = axis_coords.get(coord, None)
+    else:
+        option_value = axis_coords
+
+    if option_value is None:
+        c = da[coord]
+        if "long_name" in c.attrs:
+            label = c.long_name
+        else:
+            label = coord
+        if "units" in c.attrs:
+            label += f" [{c.units}]"
```

Comment:
Don't modify in-place to ensure we do not change c.long_name.
```suggestion
            label = label + f" [{c.units}]"
```